packages/kbn-handlebars/README.md

```diff
@@ -0,0 +1,173 @@
+
+## Development
+
+Some of the tests have been copied from the upstream `handlebars` project and modified to fit our use-case, test-suite, and conding conventions. They are all located under the `packages/kbn-handlebars/src/upstream` directory. To check if any of the copied files have received updates upstream that we might want to include in our copies, you can run the following script:
+
+```sh
+./packages/kbn-handlebars/scripts/check-for-test-changes.sh
+```
+
+If the script outputs a diff for a given file, it means that this file has been updated.
+
+Once all updates have been manually merged with our versions of the files, run the following script to "lock" us into the new updates:
+
+```sh
+./packages/kbn-handlebars/scripts/update-test-patches.sh
+```
+
+This will update the `.patch` files inside the `packages/kbn-handlebars/.patches` directory. Make sure to commit those changes.
+
+## Debugging
+
+### Print AST
+
+To output the generated AST object structure in a somewhat readable form, use the following script:
+
+```sh
+./packages/kbn-handlebars/scripts/print_ast.js
+```
+
+Example:
+
+```sh
+./packages/kbn-handlebars/scripts/print_ast.js '{{value}}'
+```
+
+Output:
+
+```js
+{
+  type: 'Program',
+  body: [
+    {
+      type: 'MustacheStatement',
+      path: {
+        type: 'PathExpression',
+        data: false,
+        depth: 0,
+        parts: [ 'value' ],
+        original: 'value'
+      },
+      params: [],
+      hash: undefined,
+      escaped: true,
+      strip: { open: false, close: false }
+    }
+  ],
+  strip: {}
+}
+```
+
+You can also filter which properties not to display, e.g:
+
+```sh
+./packages/kbn-handlebars/scripts/print_ast.js '{{#myBlock}}Hello {{name}}{{/myBlock}}' params,hash,loc,strip,data,depth,parts,inverse,openStrip,inverseStrip,closeStrip,blockParams,escaped
+```
+
+Output:
+
+```js
+{
+  type: 'Program',
+  body: [
+    {
+      type: 'BlockStatement',
+      path: { type: 'PathExpression', original: 'myBlock' },
+      program: {
+        type: 'Program',
+        body: [
+          {
+            type: 'ContentStatement',
+            original: 'Hello ',
+            value: 'Hello '
+          },
+          {
+            type: 'MustacheStatement',
+            path: { type: 'PathExpression', original: 'name' }
+          }
+        ]
+      }
+    }
+  ]
+}
+```
+
+### Environment variables
+
+By default each test will run both the original `handlebars` code and the modified `@kbn/handlebars` code to compare if the output of the two are identical. When debugging, it can be beneficial to islate a test run to just one or the other. To control this, you can use the following environment variables:
```

Comment:
```suggestion
By default each test will run both the original `handlebars` code and the modified `@kbn/handlebars` code to compare if the output of the two are identical. When debugging, it can be beneficial to isolate a test run to just one or the other. To control this, you can use the following environment variables:
```